lib/wibox/layout/ratio.lua

```diff
@@ -0,0 +1,342 @@
+---------------------------------------------------------------------------
+--- Fill all the available space. Each widget is assigned a ratio (percentage)
+-- of the total space. Multiple methods are available to ajust this ratio.
+-- @author Emmanuel Lepage Vallee
+-- @copyright 2016 Emmanuel Lepage Vallee
+-- @release @AWESOME_VERSION@
+-- @classmod wibox.layout.ratio
+---------------------------------------------------------------------------
+
+local base  = require("wibox.widget.base" )
+local flex  = require("wibox.layout.flex" )
+local fixed = require("wibox.layout.fixed")
+local table = table
+local pairs = pairs
+local floor = math.floor
+local util  = require("awful.util")
+
+local ratio = {}
+
+--- Layout a fixed layout. Each widget gets just the space it asks for.
+-- @param layout The layout you are modifying.
+-- @param context The context in which we are drawn.
+-- @param width The available width.
+-- @param height The available height.
```

Comment:
> Can I later do a PR just to fix them? 

Sure.  Although it might be forgotten probably.

I don't know about the other question.